pkg/service/handlers.go

```diff
@@ -0,0 +1,159 @@
+package service
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/gorilla/mux"
+	"github.com/moov-io/irs/pkg/config"
+	"github.com/moov-io/irs/pkg/file"
+)
+
+func parseInputFromRequest(r *http.Request) (file.File, error) {
+	src, _, err := r.FormFile("file")
+	if err != nil {
+		return nil, err
+	}
+	defer src.Close()
+
+	var input bytes.Buffer
+	if _, err = io.Copy(&input, src); err != nil {
+		return nil, err
+	}
+
+	space := regexp.MustCompile(`\s+`)
+	buf := space.ReplaceAllString(input.String(), " ")
+	mf, err := file.CreateFile([]byte(buf))
+	if err != nil {
+		return nil, err
+	}
+	return mf, nil
+}
+
+func outputString(w http.ResponseWriter, output string) {
+	w.WriteHeader(http.StatusOK)
+	w.Write([]byte(output))
+}
+
+func outputJson(w http.ResponseWriter, output interface{}) {
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(output)
+}
+
+// title: validate irs file
+// path: /validator
+// method: POST
+// produce: multipart/form-data
+// responses:
+//   200: OK
+//   400: Bad Request
+//   501: Not Implemented
+func validator(w http.ResponseWriter, r *http.Request) {
```

Comment:
Can we write an OpenAPI specification for these routes?

Reply:
yes, I will add the OpenAPI specification in next PR